caramel/module/scripts/caramel.handlebars.js

```diff
@@ -513,14 +514,54 @@ engine('handlebars', (function () {
      */
     theme = function (page, contexts, js, css, code) {
         var file, template, path, area, blocks, helper, length, i, o, areas, block,
-            areaContexts, data, areaData, find, blockData, resolve, partials, findPartials, analyzePartials,
+            areaContexts, data, areaData, find, blockData, resolve, partials, findPartials, analyzePartials, testFunction,
             theme = caramel.theme(),
             meta = caramel.meta(),
-            xcd = meta.request.getHeader(caramelData);
+            xcd = meta.request.getHeader(caramelData),
+            cct = meta.request.getHeader(CaramelCompiledTemplates);
+
         js = js || [];
         css = css || [];
         code = code || [];
 
+        /**
+         * Set the Caramel-Compiled header to compile the required partial from the backend
+         * and send the HTML block to the frontend.
+         */
+        if (cct) {
+            areas = parse(cct);
+            if (areas != null) {
+                /**
+                 * At the moment we only support one key-value pair of partials to be rendered using this function
+                 * at once. Therefore retrieve the 0th element of the array.
+                 */
+                var areaKey = Object.keys(areas)[0];
+                path = caramel.theme().resolve(partialsDir + '/' + areas[areaKey] + '.hbs');
+                if (log.isDebugEnabled()) {
+                    log.debug('Rendering page : ' + path);
+                }
+                try {
+                    file = new File(path);
+                    file.open('r');
+                    template = Handlebars.compile(file.readAll());
+                } catch (e) {
+                    log.error(e.toString());
+                    throw new Error('Error while reading the partial', e);
+                } finally {
+                    file.close();
```

Comment:
What if, try block fails before file opening, Can't close a null object. Please do a null check before closing the file.